src/syntax/ast.rs

```diff
@@ -5,101 +5,122 @@ pub struct Resource {
 }
 
 #[derive(Debug, PartialEq)]
-pub struct Comment {
-    pub body: String,
-}
-
-#[derive(Debug, PartialEq)]
-pub struct Section {
-    pub key: Key,
-    pub body: Vec<Entry>,
+pub struct Message {
+    pub id: Identifier,
+    pub value: Option<Pattern>,
+    pub attributes: Option<Vec<Attribute>>,
     pub comment: Option<Comment>,
 }
 
 #[derive(Debug, PartialEq)]
 pub enum Entry {
     Message(Message),
-    Comment(Comment),
     Section(Section),
+    Comment(Comment),
     Junk(JunkEntry),
 }
 
 #[derive(Debug, PartialEq)]
-pub struct Message {
+pub struct Pattern {
+    pub elements: Vec<PatternElement>,
```

Comment:
Do we still need `PatternElement`? Can you add `Text(String)` to `Expression` and make `elements` here be a `Vec<Expression>`?